ehsm_kms_service/secret_manager_apis.js

```diff
@@ -265,9 +360,709 @@ const createSecret = async (res, appid, payload, DB) => {
+            return
+        }
+
+        // build selector for search secret_metadata
+        let selector = {
+            appid: appid
+        }
+        if (secretName != '' && secretName != undefined) {
+            selector['secretName'] = secretName
+        }
+
+        // search secret_metadata, Maximum 4000 line 
+        const query = {
+            selector,
+            fields: ['secretName', 'description', 'createTime', 'plannedDeleteTime'],
+            limit: 4000,
+        }
+        let secret_metadata_result = await DB.partitionedFind('secret_metadata', query)
+
+        // build result
+        let result = {
+            'totalCount': 0,
+            'secretList': []
+        }
+        if (secret_metadata_result.docs.length > 0) {
+            result.totalCount = secret_metadata_result.docs.length
+            for (const doc of secret_metadata_result.docs) {
+                let secret = {};
+                secret['secretName'] = base64_decode(doc['secretName'])
+                secret['description'] = base64_decode(doc['description'])
+                secret['createTime'] = doc['createTime']
+                if (doc['plannedDeleteTime']) {
+                    secret['plannedDeleteTime'] = doc['plannedDeleteTime']
+                }
+                result.secretList.push(secret)
+            }
+        }
+
+        res.send(_result(200, 'list secrets success.', result))
+        return
+    } catch (e) {
+        console.info('listSecrets :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * Obtains the metadata of a secret.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controllerler
+ * @param {Object} payload
+ *          ==> {r}secretName(String [1~64]) : The name of the secret. eg. 'secretName01'
+ * @returns {Object}
+ *          ==> {o}secretName(String) : The name of the secret. eg. 'secretName01'
+ *          ==> {o}description(String) : The description of the secret. eg. 'desc01'
+ *          ==> {o}createTime(long) : The time when the secret was created. eg. 1659519772925
+ *          ==> {o}plannedDeleteTime(long) : The time when the secret is scheduled to be deleted. eg. 1659519772925
+ *          ==> {o}rotationInterval(String) : The interval for automatic rotation. 
+ *                                      format: integer[unit],
+ *                                      unit can be d (day), h (hour), m (minute), or s (second)
+ *                                      eg. '30d'
+ *          ==> {o}lastRotationDate(long) : The time when the last rotation was performed. eg. 1659519772925
+ *          ==> {o}nextRotationDate(long) : The time when the next rotation will be performed. eg. 1659519772925
+ */
+const describeSecret = async (res, appid, DB, payload) => {
```

Comment:
Can describe a deleted Secret? The return does not show the deleted information. Does AWS or Ali KMS also act the same?

Reply:
Yes, the logically deleted secret can be queried